common/types/hashable.go

```diff
@@ -0,0 +1,10 @@
+package types
+
+// A chain-agnostic generic interface to represent the following native types on various chains:
+// PublicKey, Address, Account, BlockHash, TxHash
+type Hashable interface {
+	ToBytes() []byte
+	ToString() string
+	FromString(string)
+	Equals(h Hashable) bool
```

Comment:
I suppose the assumption is that on eth the string will always start with `0x`? So we don't need separate `ToHex` or `FromHex` methods.

Maybe just `String()` and `UnmarshalText()` are enough since that is conformant to existing golang idioms, wdyt?

Reply:
Eth types do have Hex() and HexToAddress()/HexToHash(). However, other chains won't always have those string as hex, so these functions won't work there. Thus, the common interface doesn't list them.
In addition, the Txmgr actually never really needs the Hex() string. All it needs, is to serialize/deserialize them, compare them, etc.

I can rename to the following methods:
- String() string : converts to string
- Bytes() byte[] : convents to byte[]
- UnmarshalText(input []byte) error : parse from input, and can fail.

That sound good?

Reply:
I'm not sure I follow exactly, but we shouldn't we at least have symmetric sets of methods?

Why exactly aren't the std library text and binary marshalers appropriate?
https://pkg.go.dev/encoding

Reply:
I can use the std library text marshallers names.
I was earlier trying to align more with geth.common.Hash and geth.common.Address functions.